back/pkg/availability/availability.service.go

```diff
@@ -33,59 +32,79 @@ func NewAvailabilityService(service *AvailabilityService) *AvailabilityService {
 	}
 }
 
-func (s *AvailabilityService) Create(data *AvailabilityCreateDto, eventId uuid.UUID, user *guard.Claims) (model.Availability, error) {
-	// Acquire per-event mutex to prevent concurrent availability modifications
-	lockKey := fmt.Sprintf("%s:%s", user.Id.String(), eventId.String())
+// validateAvailabilityTimes validates the time constraints for an availability
+func (s *AvailabilityService) validateAvailabilityTimes(startsAt, endsAt time.Time, event *model.Event) error {
+	// Prevent creating/updating availabilities with end date before start date
+	if startsAt.After(endsAt) {
+		return constants.ERR_EVENT_START_AFTER_END.Err
+	}
 
-	value, _ := s.locks.LoadOrStore(lockKey, &sync.Mutex{})
-	mu := value.(*sync.Mutex)
+	// Prevent creating/updating availabilities with less than minimum duration of 5 minutes
+	minDuration := 5 * time.Minute
+	duration := endsAt.Sub(startsAt)
+	if duration < minDuration {
+		return constants.ERR_AVAILABILITY_DURATION_TOO_SHORT.Err
+	}
 
-	mu.Lock()
-	defer mu.Unlock()
+	// Prevent creating/updating availabilities not aligned on 5 minutes interval
+	// Check if times are exactly on 5-minute boundaries (no seconds or sub-seconds)
+	if startsAt.Truncate(5*time.Minute) != startsAt || endsAt.Truncate(5*time.Minute) != endsAt {
+		return constants.ERR_AVAILABILITY_INVALID_TIME_INTERVAL.Err
+	}
 
-	// Get event
-	var event model.Event
-	if err := s.eventRepository.FindOneById(eventId, &event); err != nil {
-		return model.Availability{}, constants.ERR_EVENT_NOT_FOUND.Err
+	// Prevent creating/updating availabilities outside of event date range
+	if startsAt.Before(event.StartsAt) {
+		return constants.ERR_AVAILABILITY_START_BEFORE_EVENT.Err
+	}
+	if endsAt.After(event.EndsAt) {
+		return constants.ERR_AVAILABILITY_END_AFTER_EVENT.Err
+	}
+
+	return nil
+}
+
+// validateEventAccess validates that the event exists, is accessible, and not ended
+func (s *AvailabilityService) validateEventAccess(eventId uuid.UUID, userId *uuid.UUID, event *model.Event) error {
+	if event == nil || event.Id == uuid.Nil {
+		if err := s.eventRepository.FindOneById(eventId, event); err != nil {
+			return constants.ERR_EVENT_NOT_FOUND.Err
+		}
 	}
```

Comment:
j'ai un doute sur cet enchainement de conditions.
ça devrait pas être plutôt si event == nil ou event.id == uuid.Nil ou event pas trouvé alors return not found ?
Dans ce style là : 


```suggestion
	if event == nil || event.Id == uuid.Nil || (err := s.eventRepository.FindOneById(eventId, event); err != nil) {
		return constants.ERR_EVENT_NOT_FOUND.Err
    }
```

Reply:
Oui c'est la même chose mais on ne peut pas l'écrire comme ça :
<img width="1114" height="127" alt="image" src="https://github.com/user-attachments/assets/6649db9c-db56-490c-8e97-0473bda5e3c0" />

Reply:
Ha ok 👍